### PR TITLE
meta-mbl-apps:mbl-cloud-client: Update to cloud client version 4.1.0

### PR DESCRIPTION
**Pull request by diego-sueiro**
Jenkins:
**Dev:**
http://jenkins.mbed-linux.arm.com/job/ds-zeus-dev/11/ (failed on imx8, Fix-pthread-stack-min.patch introduced to fix)
http://jenkins.mbed-linux.arm.com/job/ds-zeus-dev/12/

**Prod:**
http://jenkins.mbed-linux.arm.com/job/ds-zeus-prod/4/ (failed on imx8, Fix-pthread-stack-min.patch introduced to fix)
http://jenkins.mbed-linux.arm.com/job/ds-zeus-prod/5/

**Sync:**
https://github.com/ARMmbed/mbl-core/pull/265

### CODE REVIEW

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Index: mbed-cloud-client/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
+===================================================================
+--- mbed-cloud-client.orig/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
++++ mbed-cloud-client/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
+@@ -69,11 +69,11 @@
+ 
+ // 16KB does not seem to be enough, some tests are failing with it
+ #ifndef PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE
+-    #define PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE (1024 * 24)
++    #define PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE PTHREAD_STACK_MIN
```
[the remark this answers is not in the document]
Comment (jonathanhaigh-arm) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch]: A bit of an aside: this file should probably have an `#include <pthread.h>`.

**Comment by diego-sueiro** [comment]
@jonathanhaigh-arm , I added @teetak01 as a reviewer to provide comments.

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Index: git/cloud-services/mbl-cloud-client/mbed-cloud-client/tinycbor/src/cborvalidation.c
+===================================================================
+--- git.orig/cloud-services/mbl-cloud-client/mbed-cloud-client/tinycbor/src/cborvalidation.c
++++ git/cloud-services/mbl-cloud-client/mbed-cloud-client/tinycbor/src/cborvalidation.c
+@@ -436,7 +436,7 @@ static CborError validate_container(Cbor
+ {
+     CborError err;
+     const uint8_t *previous = NULL;
+-    const uint8_t *previous_end;
++    const uint8_t *previous_end = NULL;
```
[the remark this answers is not in the document]
Comment (teetak01) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch]: That code is owned by https://jira.arm.com/projects/IOTPCLT/

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (diego-sueiro) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch]: Addressed in be40f67
PR raised against the mainline project: https://github.com/ARMmbed/mbed-client-pal/pull/854

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (diego-sueiro) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch]: PR raised against the mainline project: https://github.com/ARMmbed/tinycbor/pull/3

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanhaigh-arm) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-pthread-stack-min.patch]: Thanks.

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonathanhaigh-arm) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Fix-tinycbor-Werror-maybe-uninitialized.patch]: Thanks.

**Comment by diego-sueiro** [comment]
Jenkins:
Dev:
http://jenkins.mbed-linux.arm.com/job/ds-zeus-dev/14/

Prod:
http://jenkins.mbed-linux.arm.com/job/ds-zeus-prod/7/

File: meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Remove-TLS-Tests.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Index: git/cloud-services/mbl-cloud-client/mbed-cloud-client/mbed-client-pal/Test/CMakeLists.txt
+===================================================================
+--- git.orig/cloud-services/mbl-cloud-client/mbed-cloud-client/mbed-client-pal/Test/CMakeLists.txt
++++ git/cloud-services/mbl-cloud-client/mbed-cloud-client/mbed-client-pal/Test/CMakeLists.txt
+@@ -59,8 +59,6 @@ set (PAL_TEST_BSP_DIR       ${CMAKE_CURR
+ set (PAL_UTILS_DIR          ${CMAKE_CURRENT_SOURCE_DIR}/../Utils/memoryProfiler/Other)
+ 
+ 
+-file(GLOB CLOUD_CREDENTIALS_SRC "${CMAKE_CURRENT_SOURCE_DIR}/../../../mbed_cloud_dev_credentials.c")
```
[the remark this answers is not in the document]
Comment (diego-sueiro) [reply meta-mbl-apps/recipes-connectivity/mbl-cloud-client/files/Remove-TLS-Tests.patch]: Addressed in be40f67 and https://github.com/ARMmbed/mbl-core/pull/265/commits/b14ef08dbc9ec70e5b105d2082f347b4d33f2d55

**Pull request merged**